### ISSUE TITLE
feat: add authentication backend

### ISSUE THREAD

**Issue by dinxsh** [itsmohit097/medi-hub#1]
add jwt based authentication (backend code)

steps i would take
1. install packages
2. create models
3. register and login routes
4. validate user credentials & generate JWT
5. middleware to verify JWT 
7. store JWT on client side and include in every req
8. logout route to remove jwt

flexible to use any other authentication methods

**Comment by AbdulhameedSk** [comment]
@itsmohit097 , please assign me this issue. I want to work on it.

**Comment by ManpreetSingh18** [comment]
I am interested in working on implementing JWT-based authentication for the backend code.I have previous experience working with JWT in earlier projects and understand its importance in ensuring secure authentication. For instance, I utilized JWT in a previous project to manage user authentication and authorization effectively. Drawing from this experience, I am confident in my ability to deliver a reliable and flexible authentication system for this project.Link of project where I have used JWT -[https://github.com/ManpreetSingh18/tavPrasad](url)
Looking forward to contributing to this issue.
